spring-cloud-gcp-starters/spring-cloud-gcp-starter-sql/src/main/java/org/springframework/cloud/gcp/sql/autoconfig/GcpCloudSqlAutoConfiguration.java

```diff
@@ -0,0 +1,118 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.sql.autoconfig;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+
+import javax.sql.DataSource;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.gax.core.CredentialsProvider;
+import com.google.api.services.sqladmin.SQLAdmin;
+import com.google.auth.http.HttpCredentialsAdapter;
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.jdbc.DataSourceAutoConfiguration;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.core.GcpProperties;
+import org.springframework.cloud.gcp.sql.CloudSqlJdbcUrlProvider;
+import org.springframework.cloud.gcp.sql.GcpCloudSqlProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+
+/**
+ * Google Cloud SQL starter.
+ *
+ * <p>It simplifies database configuration in that only an instance and database name are provided,
+ * instead of a JDBC URL. If the instance region isn't specified, this starter attempts to get it
+ * through the Cloud SQL API.
+ *
+ * @author João André Martins
+ */
+@Configuration
+@EnableConfigurationProperties(GcpCloudSqlProperties.class)
+@ConditionalOnClass({GcpProperties.class, GoogleCredential.class})
```

Comment:
OK, then `SQLAdmin` sounds like an adequate choice.

I'm also wondering if we should add `GoogleCredentials` to the list. It is kind of a transitive dependency, because `GcpProperties` is enabled by `GcpContextAutoConfiguration`, and that depends on `GoogleCredentials`. But it would probably be cautious to add it here, too. (e.g., in case `GcpContextAutoConfiguration` no longer depends on `GoogleCredentials` in the future).